src/ir/auth_logic/souffle_emitter.h

```diff
@@ -61,6 +67,13 @@ class SouffleEmitter {
   }
 
   std::string EmitPredicate(const datalog::Predicate& predicate) {
+    std::vector<std::string> comparision_operators = {"<",  ">",  "=",
+                                                      "!=", "<=", ">="};
+    if (std::find(comparision_operators.begin(), comparision_operators.end(),
+                  predicate.name()) != comparision_operators.end()) {
+      return absl::StrCat(predicate.args().front(), predicate.name(),
+                          predicate.args().back());
+    }
```

Comment:
A few chagnes:
 - comparision_operators` -> `comparison_operators`.
 - check the size of `predicate_args`.
 - Define a static array to avoid construction of the vector every time this method is called.

```suggestion
    const absl::string_view comparson_operators[] = {"<", ">", "=", "!=", "<=", ">="};
    if (std::find(std::begin(comparison_operators), std::end(comparison_operators),
                  predicate.name()) != std::end(comparison_operators)) {
      CHECK(predicate_args.size() == 2);
      return absl::StrCat(predicate.args().front(), predicate.name(),
                          predicate.args().back());
    }
```